x-pack/plugins/enterprise_search/public/applications/app_search/components/settings/log_retention/log_retention_logic.ts

```diff
@@ -0,0 +1,128 @@
+  updateLogRetention(logRetention: ILogRetention): { logRetention: ILogRetention };
+}
+
+interface ILogRetentionValues {
+  logRetention: ILogRetention | null;
+  logsRetentionUpdating: boolean;
+  openModal: ELogRetentionOptions | null;
+}
+
+export const LogRetentionLogic = kea<MakeLogicType<ILogRetentionValues, ILogRetentionActions>>({
+  path: ['enterprise_search', 'app_search', 'log_retention_logic'],
+  actions: () => ({
+    clearLogRetentionUpdating: true,
+    closeModals: true,
+    fetchLogRetention: true,
+    saveLogRetention: (option, enabled) => ({ enabled, option }),
+    setOpenModal: (option) => ({ option }),
+    toggleLogRetention: (option) => ({ option }),
+    updateLogRetention: (logRetention) => ({ logRetention }),
+  }),
+  reducers: () => ({
+    logRetention: [
+      null,
+      {
+        updateLogRetention: (previousValue, { logRetention }) => {
+          return {
+            [ELogRetentionOptions.Analytics]: {
+              ...previousValue?.[ELogRetentionOptions.Analytics],
+              ...logRetention[ELogRetentionOptions.Analytics],
+            },
+            [ELogRetentionOptions.API]: {
+              ...previousValue?.[ELogRetentionOptions.API],
+              ...logRetention[ELogRetentionOptions.API],
+            },
+          };
+        },
+      },
+    ],
+    logsRetentionUpdating: [
+      false,
+      {
+        clearLogRetentionUpdating: () => false,
+        closeModals: () => false,
+        fetchLogRetention: () => true,
+        toggleLogRetention: () => true,
+      },
+    ],
+    openModal: [
+      null,
+      {
+        closeModals: () => null,
+        saveLogRetention: () => null,
+        setOpenModal: (_, { option }) => option,
+      },
+    ],
+  }),
+  listeners: ({ actions, values }) => ({
+    fetchLogRetention: async () => {
+      try {
+        const { http } = HttpLogic.values;
+        const response = await http.get('/api/app_search/log_settings');
+        actions.updateLogRetention(
+          convertLogRetentionFromServerToClient(response as ILogRetentionServer)
+        );
+      } catch (e) {
+        flashAPIErrors(e);
+      } finally {
+        actions.clearLogRetentionUpdating();
+      }
+    },
+    saveLogRetention: async ({ enabled, option }) => {
+      const updateData = { [option.toString()]: { enabled } };
+
+      try {
+        const { http } = HttpLogic.values;
+        const response = await http.put('/api/app_search/log_settings', {
+          body: JSON.stringify(updateData),
+        });
+        actions.updateLogRetention(
+          convertLogRetentionFromServerToClient(response as ILogRetentionServer)
+        );
+      } catch (e) {
+        flashAPIErrors(e);
+      } finally {
+        actions.clearLogRetentionUpdating();
+      }
+    },
+    toggleLogRetention: ({ option }) => {
+      const logRetention = values.logRetention && values.logRetention[option];
```

Comment:
We can simplify this to:

```suggestion
      const logRetention = values.logRetention?.[option];
```